### ISSUE TITLE
Support for Gitlab MR's on Private instances

### ISSUE THREAD

**Issue by ig12** [akuity/kargo#2079]
# Proposed Feature

Support the PullRequest promotion Feature for custom Gitlab deployments.

# Motivation

A lot of enterprises are running their git instances. The Promotion with merge requests is a valuable feature, especially in enterprise use cases, where each change to a production environment has to be validated and approved. The Merge Request Process is often already implemented and can be reused but simplified thanks to Kargo.

# Suggested Implementation

I have no suggestion for the implementation of the logic itself, but in the Stage CR it could be implemented like this:

```
  promotionMechanisms:
    gitRepoUpdates:
    - pullRequest:
        gitlab:
          customUrl: gitlab.enterprise.com
      repoURL: https://gitlab.enterprise.com/playground/kargo-demo-app.git
      writeBranch: stage/prod
```

**Comment by krancour** [comment]
@ig12 the existing gitlab MR is supposed to already work for any repo URL that has `gitlab` anywhere within it.

https://github.com/akuity/kargo/blob/2d4aaf579625fd723ea4b3941c226ec9bea23d5e/internal/gitprovider/gitlab/gitlab.go#L19-L29

Have you tried an found this not to be working?

**Comment by ig12** [comment]
Yes, I have tried it and it did not work. But this is because our URL is https://code.enterprise.com.
I would rephrase the request to support gitlab instances that do not have gitlab inside the URL.

**Comment by krancour** [comment]
@ig12 it looks like there are already existing fields that allow one to hint at what git provider that pull request-based PR should be working with and we're just not looking at those fields when selecting a provider.

This should be a somewhat light lift, I think.
